e2e/timer-reset-mocks/after-reset-all-mocks/timerAndMock.test.js

```diff
@@ -4,17 +4,17 @@ describe('timers', () => {
   it('should work before calling resetAllMocks', () => {
     jest.useFakeTimers();
     const f = jest.fn();
-    setImmediate(() => f());
-    jest.runAllImmediates();
-    expect(f.mock.calls.length).toBe(1);
+    setTimeout(f, 0);
```

Comment:
non-related changes - I just kept them from the original PR since this works for both implementations 🙂